contracts/cw1-subkeys/src/contract.rs

```diff
@@ -505,6 +650,162 @@ mod tests {
         );
     }
 
+    #[test]
+    fn query_permissions_works() {
+        let mut deps = mock_dependencies(20, &[]);
+
+        let owner = HumanAddr::from("admin0001");
+        let admins = vec![owner.clone()];
+
+        // spender1 has every permission to stake
+        let spender1 = HumanAddr::from("spender0001");
+        // spender2 do not have permission
+        let spender2 = HumanAddr::from("spender0002");
+        // non existent spender
+        let spender3 = HumanAddr::from("spender0003");
+
+        let god_mode = Permissions {
+            delegate: true,
+            redelegate: true,
+            undelegate: true,
+            withdraw: true,
+        };
+
+        let env = mock_env(owner.clone(), &[]);
+        // Init a contract with admins
+        let init_msg = InitMsg {
+            admins: admins.clone(),
+            mutable: true,
+        };
+        init(&mut deps, env.clone(), init_msg).unwrap();
+
+        let setup_perm_msg1 = HandleMsg::SetPermissions {
+            spender: spender1.clone(),
+            permissions: god_mode,
+        };
+        handle(&mut deps, env.clone(), setup_perm_msg1).unwrap();
+
+        let setup_perm_msg2 = HandleMsg::SetPermissions {
+            spender: spender2.clone(),
+            // default is no permission
+            permissions: Default::default(),
+        };
+        // default is no permission
+        handle(&mut deps, env.clone(), setup_perm_msg2).unwrap();
+
+        let permissions = query_permissions(&deps, spender1.clone()).unwrap();
```

Comment:
Looks good. I would add one more test, where you set the allowance for someone with a permission and then ensure that both are set properly (setting allowance doesn't unset the permission)